### PR TITLE
Domain Suggestions: Add signup ab test

### PR DESCRIPTION
**Pull request by matthusby**
This is a mirror of the test run on tortuga here:

p8kIbR-c9-p2

### CODE REVIEW

**Comment by matthusby** [comment]
e2e test added in https://github.com/Automattic/wp-e2e-tests/pull/913

File: client/components/domains/register-domain-step/index.jsx [hunk 1]
```diff
@@ -65,10 +66,16 @@ let lastSearchTimestamp = null;
 let searchCount = 0;
 let recordSearchFormSubmitWithDispatch;
 
+const testGroup = abtest( 'domainSuggestionTestV5' );
```
Comment (klimeryk) [inline client/components/domains/register-domain-step/index.jsx]: Just a sanity check - won't this cause the user to get assigned a variation too early?

Comment (delputnam) [reply client/components/domains/register-domain-step/index.jsx]: Yes, @klimeryk. Thanks! @matthusby I totally missed this. This (and the following `if`) should be done inside the class.

**Comment by klimeryk** [comment]
Just noting for future reference that this broke domain search and had to be reverted: https://github.com/Automattic/wp-calypso/pull/22205

**Pull request merged**